### PR TITLE
Initial draft of R-intro assignments

### PR DESCRIPTION
**Pull request by joelostblom**
Not final, but please provide opinions!

Note that these assignments can be modified based on what prerequisite knowledge is needed for @mbonsma 's lectures.

The dplyr part is largely taken from the data carpentry R ecology assignments.

### CODE REVIEW

File: course/assignments/R-intro-lectures-assignments.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+5. Data frames
+    a. What are the column names of inbuilt dataframe `iris`?
+    b. How can you view the first 5 rows of this data frame?
+    c. How many observations are there?
+    d. Which factor levels are there? What is a factor level and when can it be useful?
+    e. What is the mean value for the sepal.height column?
+6. Save the data frame with only the species and sepal.width columns as csv-file. Load this csv file as a new data frame and run display its summary statistics.
+
+
+# Week 3 (8 marks) dplyr, ggplot
```
Comment (lwjohnst86) [inline course/assignments/R-intro-lectures-assignments.md]: This will probably be a bit hard as an early assignment, especially for only 8 marks and given that it requires some conceptual thinking about data and wrangling it that the students might not have yet.

Comment (lwjohnst86) [inline course/assignments/R-intro-lectures-assignments.md]: Some suggestions/options to consider. Either:

- Have few questions but worded in plain english about what we want the output to look like (e.g. the second question)
- Or, have more questions, but worded closer to how the R commands would be worded and ordered (e.g. question 3.b.)

It depends on what we want them to learn from the assignment. Do we want them to learn the commands first (it is the 3rd week) or do we want them to learn how to convert a human goal/thought and convert it into computer terms?

File: course/assignments/R-intro-lectures-assignments.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+1. Using two separate statements, select the columns `plot_id`, `species_id`, and `weight` from the `surveys` dataframe, and filter it to only include data from the year 1995. Do the same thing using pipes (%>%).
+2. Create a new data frame from the surveys data that meets the following criteria: Contains only the species_id column and a new column called hindfoot_half containing values that are half the hindfoot_length values. In this  hindfoot_half column, there are no NAs and all values are less than 30.
+3. Basic data aggregation.
+    a. How many individuals were caught in each plot_type surveyed?
+    b. Use group_by() and summarize() to find the mean, min, and max hindfoot length for each species (using species_id).
+    c. What was the heaviest animal measured in each year? Return the columns  year, genus, species_id, and weight.
+    d. Count the number of individuals of each sex.
+
+4. Reshape data.
```
Comment (lwjohnst86) [inline course/assignments/R-intro-lectures-assignments.md]: This question would be more useful in a later assignment. This requires some pretty challenging ways of thinking about data that the students will likely not have just yet, especially for the amount of grades this is worth. (maybe they will have the understanding.. we could always put it as a "bonus grade" or something)

File: course/assignments/R-intro-lectures-assignments.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    b. Now take that data frame, and make it long again, so each row is a unique  plot_id year combination.
+    c. The surveys data set is note truly wide or long because both there are two columns of measurement - hindfoot_length and weight. This makes it difficult to do things like look at the relationship between mean values of each measurement per year in different plot types. Let’s walk through a common solution for this type of problem. First, use gather to create a truly long dataset where we have a key column called measurement and a value column that takes on the value of either hindfoot_length or weight. Hint: You’ll need to specify which columns are being gathered.
+    d. With this new truly long data set, calculate the average of each measurement in each year > for each different plot_type. Then spread them into a wide data set with a column for hindfoot_length and weight. Hint: Remember, you only need to specify the key and value columns for spread.
+5. Plotting
```
Comment (lwjohnst86) [inline course/assignments/R-intro-lectures-assignments.md]: I like this one. Plus, who doesn't like to visualize things?? :D

**Comment by joelostblom** [comment]
@lwjohnst86 Thanks for all the comments on this and sorry for being slow to get back to you. I will make changes and upload a new version in a few days when I am back at my computer.

File: course/assignments/R-intro-lectures-assignments.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# Week 1 (4 marks) Intro to programming
```
Comment (joelostblom) [inline course/assignments/R-intro-lectures-assignments.md]: Yes, lets make a survey. I make task 4 filling out the survery and move the current 4 into the survey

File: course/assignments/R-intro-lectures-assignments.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+1. Why is it beneficial to use a programming language rather than a spreadsheet software for data analyses?
+2. Get setup with Rstudio at home (or a lab computer after hours). Install R, Rstudio, and the necessary packages (tidyverse, rmarkdown). Use `install.packages(<package_name>)`. Load all the libraries you installed into your environment with `library(<package_name>)` and send us the output of running `sessionInfo()`.
```
Comment (joelostblom) [inline course/assignments/R-intro-lectures-assignments.md]: Reworded. Do you think it would be more useful for them to learn how to make a pdf? Sicne Rmd and md are so close in format anyways. I guess I can just go through that in the first lesson, should be fast.

Comment (joelostblom) [inline course/assignments/R-intro-lectures-assignments.md]: I also think one of the fun things with markdown is to generate professional looking pdfs easily, so it might be an encouraging task to include =)

File: course/assignments/R-intro-lectures-assignments.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    c. Extract every 5th element from `v`, into a new vector `v_every_5th`.
+4. Boolean indexing
+    a. Create a boolean vector `v_bool` of the same length as `v`, indicating which vector elements are bigger than 20.
+    b. Use `v_bool` as an index to extract the elements from `v` that are bigger than 20 and save as `v_over_20`.
+    c. What's the mean of `v_over_20`?
+5. Data frames
+    a. What are the column names of inbuilt dataframe `iris`?
+    b. How can you view the first 5 rows of this data frame?
+    c. How many observations are there?
+    d. Which factor levels are there? What is a factor level and when can it be useful?
```
Comment (joelostblom) [inline course/assignments/R-intro-lectures-assignments.md]: I think I will remove this, it is not crucial to understand what a factor level is. I think understanding how to use `group_by` is sufficient, and since it can make sense to group by integers as well, it might just be confusing to introduce factor levels.

File: course/assignments/R-intro-lectures-assignments.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
+# Week 1 (4 marks) Intro to programming
+1. Why is it beneficial to use a programming language rather than a spreadsheet software for data analyses?
+2. Get setup with Rstudio at home (or a lab computer after hours). Install R, Rstudio, and the necessary packages (tidyverse, rmarkdown). Use `install.packages(<package_name>)`. Load all the libraries you installed into your environment with `library(<package_name>)` and send us the output of running `sessionInfo()`.
+3. What is Rmarkdown and why are we using it in this class? How do you execute a chunk of code in an Rmarkdown document?
+4. Is there anything in particular you are excited about learning about theoretical ecological population models and about using R for data analyses and modelling?
+
+# Week 2 (8 marks) Rmarkdown, assignment, functions, vectors, data frames
```
Comment (joelostblom) [inline course/assignments/R-intro-lectures-assignments.md]: Yes, this is a good idea! I am making a `.Rmd` with all the questions and answers. Then we can include all the structure and even some of the answers if we think they students might need a few extra hints on the tougher questions.

File: course/assignments/R-intro-lectures-assignments.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+1. Assignment
+    a. Assign the value `5` to the variable/object `a`.
+    b. Assign the result of `10/3` to the variable `b`.
+    c. Assign the product of `a` and `b` to `c`.
+    d. Write a function that multiplies two numbers and returns the result. Use it to assign the product of `a` and `b` to `c`.
+2. Vectors
+    a. Create a vector `v` with all integers 1-30.
+    b. Use `v` to create a new vector `v_square` with the square of element 3, 6, 7, 10, 15, 22, 23, 24, and 30 from `v`. Do _not_ use a for loop.
+    c. Extract every 5th element from `v`, into a new vector `v_every_5th`.
+4. Boolean indexing
```
Comment (joelostblom) [inline course/assignments/R-intro-lectures-assignments.md]: But this is so important! All their filtering later builds upon a very basic understanding of this at least. I will try to make it so that it is as intuitive as using `filter()`. I agree that we don't need to get into logic edge cases, that confuses me as well. Just basic, how many values are over 20 kind of thing.

File: course/assignments/R-intro-lectures-assignments.md (hunk 5 above)
Comment (lwjohnst86) [inline course/assignments/R-intro-lectures-assignments.md]: It would be nice, however... that would require installing MikTeX or whatever equivalent is available. This may or may not be too much work (installing LaTeX can sometimes be a major pain). Are we able to get LaTeX installed on the computers beforehand? We'll have to mention that if they do this on their own laptops or whatever, that it might not work/take a lot of effort to install.

Comment (joelostblom) [inline course/assignments/R-intro-lectures-assignments.md]: Good point, I forgot about that LaTeX needs to be installed... I haven't done it on Windows myself, but it [seems straightforward](https://medium.com/@sorenlind/create-pdf-reports-using-r-r-markdown-latex-and-knitr-on-windows-10-952b0c48bfa9), just install MikTeX as you mentioned.

I will follow up in #50 and see if we can get these installed. I will try it first to see which R-latex packages are needed. I think this is important since a strong point for me to use markdown in the first place is that I can easily render PDFs and hand in for any class assignment. For this purpose, it is not as useful to teach how to generate markdown documents, since teachers will probably not accept text files for assignments.

**Comment by lwjohnst86** [comment]
Closing as it has been moved over into `rcourse-assignments` repo.
